feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,118 @@
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine._base_transformers.base_numerical import BaseNumericalTransformer
+from feature_engine._base_transformers.mixins import (
+    FitFromDictMixin,
+    GetFeatureNamesOutMixin,
+)
+from feature_engine._check_input_parameters.check_init_input_params import (
+    _check_param_drop_original,
+)
+
+
+class DistanceFeatures(
+    BaseNumericalTransformer, FitFromDictMixin, GetFeatureNamesOutMixin
+):
+    EARTH_RADIUS: float = 6371.  # radius of Earth in kms
+
+    def __init__(
+            self,
+            a_latitude: str,
+            a_longitude: str,
+            b_latitude: str,
+            b_longitude: str,
+            output_column_name: Union[str, None] = None,
+            drop_original: bool = False,
+    ) -> None:
+
+        self.a_latitude = self._check_column_name(a_latitude)
```

Comment:
I'd probably just assign whatever the user passes. We need to check if it is a string or a list only, and we have that function ready to import.

Reply:
ok... but I don't understand. Should we allow a column name to be an integer?

Because you wrote:

> We allow integers as column names. I think I introduced this because it helps accommodate some of the compatibility tests with sklearn. In short, I don't think we need this check. We "outsource" this to pandas.

Therefore, I should checks that what we get is a list of str or integer, not?

Reply:
We have some example checks here:

https://github.com/feature-engine/feature_engine/blob/77ea405f18372bf3bcd8ef566bc7c3df5d66a056/feature_engine/creation/relative_features.py#L140-L157